core/lib/stores.ts

```diff
@@ -11,6 +11,11 @@ type StoreValues<Stores extends ReadableAtom[]> = {
   [Index in keyof Stores]: StoreValue<Stores[Index]>
 }
 
+/** use only for test mocks or don't permit rewrite store at all */
+export const isWritableAtom = <Value>(
```

Comment:
If you want to avoid `/* ts-expect-error */` let’s create some method like `forceSet<Value>(store: ReadonlyAtom<Value>, value: Value)`

Right now it is hard to understand why we are checking `isWritableAtom`, because other developer will think that sometimes store is writable, sometimes not.

Reply:
Can I replace the readonlyExport function everywhere with a simple type casting?
I think function a little confusing, like it should delete method `.set()` from store, but it's just ts util

Reply:
> Can I replace the readonlyExport function everywhere with a simple type casting?

No. We use types to check that API is used in correct way.

Using `.set()` for stores with `readonlyExport` could break their logic. It is OK for stories (where we do only a snapshot), but will be bad in client or test.

Reply:
No, types will remain.

I mean something like this
```diff
-export const slowPosts = readonlyExport($posts)
+export const slowPosts: ReadableAtom<SlowPostsValue> = $posts
```
or
```diff
-export const slowPosts = readonlyExport($posts)
+export const slowPosts: MakeReadOnly<typeof $posts> = $posts
```

Same typecasting. But without extra function in vanilla

Reply:
It was too long, I added helper to make it shorter.

But I understand you, functions should change JS, not types. Good point.

Let’s keep `readonlyExport` for this PR, I will think about it.